Ghidra/Processors/STM/data/languages/ST7.sinc

```diff
@@ -0,0 +1,758 @@
+  S = A;
+  nzFlags(A);
+}
+
+:LD X, S is $(NO_PRE) & opcode=0x96 & X & S {
+  X = S;
+  nzFlags(S);
+}
+
+:LD S, X is $(NO_PRE) & opcode=0x94 & X & S {
+  S = X;
+  nzFlags(X);
+}
+
+:LD Y, S is $(PRE_PDY) & opcode=0x96 & Y & S {
+  Y = S;
+  nzFlags(S);
+}
+
+:LD S, Y is $(PRE_PDY) & opcode=0x94 & Y & S {
+  S = Y;
+  nzFlags(Y);
+}
+
+:MUL X, A is $(NO_PRE) & opcode=0x42 & X & A {
+  H = 0;
+  C = 0;
+  local destination:2 = zext(X);
+  destination = destination * zext(A);
+  local low:2 = destination & 0xFF;
+  local high:2 = (destination >> 8) & 0xFF;
+  A = low:1;
+  X = high:1;
+}
+
+:MUL Y, A is $(PRE_PDY) & opcode=0x42 & Y & A {
+  H = 0;
+  C = 0;
+  local destination:2 = zext(Y);
+  destination = destination * zext(A);
+  local low:2 = destination & 0xFF;
+  local high:2 = (destination >> 8) & 0xFF;
+  A = low:1;
+  Y = high:1;
+}
+
+:NEG BITOP is kind=0x00 ... & BITOP {
+  local value = (BITOP ^ 0xFF) + 1;
+  nzFlags(value);
+  BITOP = value;
+  if !Z goto inst_next;
+  C = 0;
+}
+
+:NOP is $(NO_PRE) & opcode=0x9D {
+}
+
+:OR A, ASMEMOP is kind=0x0A ... & ASMEMOP & A {
+  local value = ASMEMOP;
+  value = value | A;
+  nzFlags(value);
+  A = value;
+}
+
+:POP A is $(NO_PRE) & opcode=0x84 & A {
+  SP = SP + 1;
+  A = *:1 SP;
+}
+
+:POP X is $(NO_PRE) & opcode=0x85 & X {
+  SP = SP + 1;
+  X = *:1 SP;
+}
+
+:POP Y is $(PRE_PDY) & opcode=0x85 & Y {
+  SP = SP + 1;
+  Y = *:1 SP;
+}
+
+:POP CC is $(NO_PRE) & opcode=0x86 & CC {
+  SP = SP + 1;
+  CC = *:1 SP;
+  C = CC[0,1];
+  Z = CC[1,1];
+  N = CC[2,1];
+@ifdef ST70
+  I = CC[3,1];
+@endif
+@ifdef ST72
+  I0 = CC[3,1];
+@endif
+  H = CC[4,1];
+@ifdef ST72
+  I1 = CC[5,1];
+@endif
+}
+
+:PUSH A is $(NO_PRE) & opcode=0x88 & A {
+  *:1 SP = A;
+  SP = SP - 1;
+}
+
+:PUSH X is $(NO_PRE) & opcode=0x89 & X {
+  *:1 SP = X;
+  SP = SP - 1;
+}
+
+:PUSH Y is $(PRE_PDY) & opcode=0x89 & Y {
+  *:1 SP = Y;
+  SP = SP - 1;
+}
+
+:PUSH CC is $(NO_PRE) & opcode=0x8A & CC {
+  CC = 0xFF;
+  CC[0,1] = C;
+  CC[1,1] = Z;
+  CC[2,1] = N;
+@ifdef ST70
+  CC[3,1] = I;
+@endif
+@ifdef ST72
+  CC[3,1] = I0;
+@endif
+  CC[4,1] = H;
+@ifdef ST72
+  CC[5,1] = I1;
+@endif
+  *:1 SP = CC;
+  SP = SP - 1;
+}
+
+:RCF is $(NO_PRE) & opcode=0x98 {
+  C = 0;
+}
+
+:RET is $(NO_PRE) & opcode=0x81 {
+  SP = SP + 1;
+  PCH = *:1 SP;
+  SP = SP + 1;
+  PCL = *:1 SP;
+}
+
+:RIM is $(NO_PRE) & opcode=0x9A {
+@ifdef ST70
+  I = 0;
+@endif
+@ifdef ST72
+  I0 = 0;
+  I1 = 1;
+@endif
+}
+
+:RLC BITOP is kind=0x09 ... & BITOP {
+  local tmpC = C;
+  local value = BITOP;
+  C = value >> 7;
+  local result = value << 1;
+  result = result | tmpC;
+  nzFlags(value);
+  BITOP = result;
+}
+
+:RRC BITOP is kind=0x06 ... & BITOP {
+  local tmpC = C << 7;
+  local value = BITOP;
+  C = value & 1;
+  local result = value >> 1;
+  result = result | tmpC;
+  nzFlags(value);
+  BITOP = result;
+}
+
+:RSP is $(NO_PRE) & opcode=0x9C {
+  SP = 0x01FF;
+}
+
+:SBC A, ASMEMOP is kind=0x02 ... & ASMEMOP & A {
```

Comment:
might see if `sborrow` would help here for the C flag

Reply:
This is taken from the [6502 definition](https://github.com/NationalSecurityAgency/ghidra/blob/0e3da48afd449d7459ee4fcf0d72067445bd6b27/Ghidra/Processors/6502/data/languages/6502.slaspec#L68) implementation, actually - I was going for something that is already tested elsewhere.  I haven't really looked at `sborrow`, I'll check that out and maybe fix the 6502/65c02 definitions too if that works.

Reply:
i didn't look into too much, just saw you were doing a lot of math, if another processor is doing it and no issue there, maybe nvmd.